cas-server-support-oauth/src/main/java/org/jasig/cas/support/oauth/web/BaseOAuthWrapperController.java

```diff
@@ -106,24 +107,28 @@ protected OAuthWebApplicationService createService(final RegisteredService regis
      * Create an authentication from a user profile.
      *
      * @param profile the given user profile
+     * @param service the registered service
      * @return the built authentication
      */
-    protected Authentication createAuthentication(final UserProfile profile) {
```

Comment:
I reviewed this briefly. Let me ask:

When OAuth creates an access token, that token is then submitted to the profile controller. The token needs to be validated, and the token is just an ST correct?

So we end up doing this:

```java
final AccessToken accessTokenTicket = this.ticketRegistry.getTicket(accessToken, AccessToken.class);
if (accessTokenTicket == null || accessTokenTicket.isExpired()) {
    //bad ticket...
}

writeOutProfileResponse(jsonGenerator, accessTokenTicket.getAuthentication().getPrincipal());
```

So this I think is the issue here. Let me elaborate:

1. This ticket/token is validated manually. The code is directly cross checked against the ticket registry, which is a practice we should discourage.
2. As you said, the principal that is used here is established during authentication as part of that access token. This is contrary to the CAS model where we always use the principal constructed during "validation". 

My theory is:

Could we submit the access token to the `/cas/p3/serviceValidate` via the Java CAS client, validate it as a normal ST, get back a CAS principal and then use that? The validation logic in CASImpl accounts for everything that is required, including attribute release, access strategy (which is what this PR is trying to do) and everything else, which should allow you to remove that `if` block up there.

Makes sense? 

P.S Not opposed to the PR, just saying there might be better ways to handle this.